compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -460,7 +462,7 @@ object SymDenotations {
       info match
         case TypeAlias(alias) if isOpaqueAlias && owner.isClass =>
           setAlias(alias)
-          HKTypeLambda.boundsFromParams(alias.typeParams, bounds(rhs))
+          HKTypeLambda.boundsFromParams(tparams, bounds(rhs))
```

Comment:
Just curious, we have to use `tparams` instead of `alias.typeParams` due to the following test in `boundsFromParams`?

```Scala
      def isOpaqueAlias = params match
        case (param: Symbol) :: _ => param.owner.is(Opaque)
        case _ => false
```

Reply:
`alias.typeParams` is a list of `TypeParamRefs`, which are different from the symbols on `rhs`. That caused the problem.